Packs/ConcentricAI/Integrations/ConcentricAI/ConcentricAI.py

```diff
@@ -0,0 +1,630 @@
+    grp_users = answers['total_by_permitted_grp_users']
+    grp_users = grp_users['buckets']
+    orphan_users = answers['total_by_permitted_orphans']
+    orphan_users = orphan_users['buckets']
+
+    result = {
+        'internal-users': int_users,
+        'external_users': ext_users,
+        'group_users': grp_users,
+        'orphan_users': orphan_users
+    }
+    return result
+
+
+def transform_user_details(answer):
+    target = {}
+    if answer['path'] is not None:
+        target['file-path'] = answer['path']
+    if 'permitted_ext_users' in answer and answer['permitted_ext_users'] is not None:
+        target['user-external'] = answer['permitted_ext_users']
+    if 'permitted_grp_users' in answer and answer['permitted_grp_users'] is not None:
+        target['users-group'] = answer['permitted_grp_users']
+    if 'name' in answer and answer['name'] is not None:
+        target['file-name'] = answer['name']
+    if 'permitted_int_users' in answer and answer['permitted_int_users'] is not None:
+        target['user-internal'] = answer['permitted_int_users']
+    if 'permitted_orphans' in answer and answer['permitted_orphans'] is not None:
+        target['users-orphans'] = answer['permitted_orphans']
+    return target
+
+
+def transform_file_information(target: dict, risk_dict: dict, queryClient: QueryClient, loginClient: LoginClient):
+
+    if target['risk_id'] is not None:
+        target['risk_names'] = get_rule_names(target['risk_id'], risk_dict)
+    if 'risk_id' in target:
+        target.pop('risk_id')
+    if 'category' in target:
+        target.pop('category')
+    if 'duplicate_contrib_cids' in target:
+        target.pop('duplicate_contrib_cids')
+    if 'word_cloud' in target:
+        target.pop('word_cloud')
+    if 'confidence' in target:
+        target.pop('confidence')
+    if 'ownerDetails' in target:
+        if target['ownerDetails'] is not None:
+            target['ownerDetails'] = target['ownerDetails']['name']
+    if 'near_duplicate_contrib_cids' in target:
+        if target['near_duplicate_contrib_cids'] is not None:
+            file_names: list = []
+            for cid in target['near_duplicate_contrib_cids']:
+                res = queryClient.get_file_names(loginClient, cid)
+                if res is not None:
+                    path = res['path']
+                    name = res['name']
+                    file_names.append(path + " --> " + name)
+            target['near_duplicate_files'] = file_names
+            target.pop('near_duplicate_contrib_cids')
+    return target
+
+
+def test_module(client: LoginClient):
+    token = client.fetch_api_token()
+    if token:
+        return 'ok'
+
+
+def fetch_incidents(loginClient: LoginClient, queryClient: QueryClient, last_run: Dict[str, int], max_results: int):
+    global SCROLL_ID_INCIDENT
+    last_fetch = last_run.get('last_fetch', None)
+    scroll_id = last_run.get('scroll_id', None)
+    if last_fetch is None:
+        from_time = 0
```

Comment:
Please implement `First Run` parameter and use it here. Recommended using`parse_date_range` in CommonServerPython.

Reply:
I am not exactly aware of the first run parameter ? any example I can refer to ?

Reply:
You can have a look in `Akami_SIEM` integration[ (specifically here)](https://github.com/demisto/content/blob/fe8e7fba0856644f7d55f0eab1d8e6578cbd0796/Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py#L272) but basically it would work as follows:
The user inputs a first_fetch time in plain text such as "3 days" or "2 months". You use the `parse_date_range` function to parse that into time back in the format you want.

Reply:
done